src/audio/mixer.c

```diff
@@ -372,6 +357,31 @@ static int mixer_copy(struct comp_dev *dev)
 		sources[i] = buffer_release(sources[i]);
 	}
 
+	if (!num_mix_sources || (frames == 0 && md->sources_inactive)) {
+		/*
+		 * Generate silence when sources are inactive. When
+		 * sources change to active, additionally keep
+		 * generating silence until at least one of the
+		 * sources start to have data available (frames!=0).
+		 */
+		frames = audio_stream_get_free_frames(&sink->stream);
+		sink_bytes = frames * audio_stream_frame_bytes(&sink->stream);
+		sink = buffer_release(sink);
+		if (!audio_stream_set_zero(&sink->stream, sink_bytes)) {
+			buffer_stream_writeback(sink, sink_bytes);
+			comp_update_buffer_produce(sink, sink_bytes);
+		}
+
+		md->sources_inactive = true;
+
+		return 0;
+	}
+
+	if (md->sources_inactive) {
+		md->sources_inactive = false;
+		comp_info(dev, "mixer_copy exit sources_inactive state");
```

Comment:
nitpick: is this `comp_info()` really needed? It was useful for debugging, but once you've verified that it works as expected, wouldn't a `comp_dbg()` be enough?

Reply:
Hmm, in original version sources_inactive state was more limited, so I thought info would be useful to track this in CI, but now this more common so comp_dbg is more appropriate. Changed version uploaded. Thanks @lyakh !